demisto_sdk/commands/common/clients/__init__.py

```diff
@@ -0,0 +1,153 @@
+import os
+from functools import lru_cache
+from typing import Optional
+
+import demisto_client
+from demisto_client.demisto_api.rest import ApiException
+from packaging.version import Version
+
+from demisto_sdk.commands.common.clients.configs import (
+    XsiamClientConfig,
+    XsoarClientConfig,
+    XsoarSaasClientConfig,
+)
+from demisto_sdk.commands.common.clients.xsiam.xsiam_api_client import XsiamClient
+from demisto_sdk.commands.common.clients.xsoar.xsoar_api_client import XsoarClient
+from demisto_sdk.commands.common.clients.xsoar_saas.xsoar_saas_api_client import (
+    XsoarSaasClient,
+)
+from demisto_sdk.commands.common.constants import (
+    AUTH_ID,
+    DEMISTO_BASE_URL,
+    DEMISTO_KEY,
+    MINIMUM_XSOAR_SAAS_VERSION,
+    MarketplaceVersions,
+)
+
+
+@lru_cache
+def get_client_from_config(
+    client_config: XsoarClientConfig, verify_ssl: bool = False
+) -> XsoarClient:
+    """
+    Returns the correct Client (xsoar on prem, xsoar saas or xsiam) based on the clients config object
+
+    Args:
+        client_config: clients configuration
+        verify_ssl: whether in each request SSL should be verified, True if yes, False if not
+
+    Returns:
+        the correct api clients based on the clients config
+    """
+    base_url = client_config.base_api_url
+    api_key = client_config.api_key
+    auth_id = client_config.auth_id
+
+    _client = demisto_client.configure(
+        base_url=base_url,
+        api_key=api_key.get_secret_value(),
+        auth_id=auth_id,
+        verify_ssl=verify_ssl,
+    )
+
+    if isinstance(client_config, XsiamClientConfig):
+        return XsiamClient(client=_client, config=client_config)
+    elif isinstance(client_config, XsoarSaasClientConfig):
+        return XsoarSaasClient(client=_client, config=client_config)
+    else:
+        return XsoarClient(client=_client, config=client_config)
+
+
+def get_client_from_marketplace(
```

Comment:
yes, for example in upload, we do not know to which environment the content items are being uploaded, this can be useful for graph operations that calculates stuff using marketplace in case anything is needed from the api